cli_tools/gce_vm_image_import/importer/importer.go

```diff
@@ -115,11 +116,19 @@ func validateSourceFile(storageClient domain.StorageClientInterface, sourceBucke
 	}
 	defer rc.Close()
 
+	byteCountingReader := daisycommon.NewByteCountingReader(rc)
 	// Detect whether it's a compressed file by extracting compressed file header
-	if _, err = gzip.NewReader(rc); err == nil {
+	if _, err = gzip.NewReader(byteCountingReader); err == nil {
 		return daisy.Errf("cannot import an image from a compressed file. Please provide a path to an uncompressed image file. If the compressed file is an image exported from Google Compute Engine, please use 'images create' instead")
 	}
 
+	// By calling gzip.NewReader above, a few bytes were read from the Reader in
+	// an attempt to decode the compression header. If the Reader represents
+	// an empty file, then BytesRead will be zero.
+	if byteCountingReader.BytesRead <= 0 {
```

Comment:
Does this work for non-zip files as well?

Reply:
Very good question -- it does work, since that gzip.NewReader constructor reads bytes from the stream no matter what. In part to figure out whether it's compressed at all.

Good news: Here's a test that confirms this assumption: https://github.com/GoogleCloudPlatform/compute-image-tools/blob/master/cli_tools/gce_vm_image_import/importer/importer_test.go#L192